contrib/pg_tde/src/catalog/tde_principal_key.c

```diff
@@ -249,22 +249,20 @@ set_principal_key_with_keyring(const char *key_name, const char *provider_name,
 		}
 	}
 
-	if (keyInfo != NULL && ensure_new_key)
+	if (!keyInfo)
 	{
-		ereport(ERROR,
-				errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-				errmsg("cannot to create key \"%s\" because it already exists", key_name));
+		if (providerOid == GLOBAL_DATA_TDE_OID)
+			ereport(ERROR,
+					errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					errmsg("key \"%s\" does not exist", key_name),
+					errhint("Use pg_tde_create_key_using_global_key_provider() to create it."));
+		else
+			ereport(ERROR,
+					errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					errmsg("key \"%s\" does not exist", key_name),
+					errhint("Use pg_tde_create_key_using_database_key_provider() to create it."));
 	}
 
-	if (strlen(key_name) >= sizeof(keyInfo->name))
```

Comment:
Don't we actually still need this check due to our file format? Imagine if a key exists which they have created themselves which is longer than the field size in our file format?